### PR TITLE
Update GCM (gnome-connections-manager) to 1.2.0~HEAD that works in current OI

### PR DESCRIPTION
**Pull request by jimklimov**
It all started with an update... and my GCM no longer started, claiming "You must install libvte for python" for reasons I could not discern (unlike https://github.com/mjun/gnome-connection-manager/issues/14 the libs are still present on my January's rootfs and the current one).

I discovered that just recently development of GCM moved to, and resumed on, github at https://github.com/kuthulux/gnome-connection-manager - proceeding to version 1.2.0 with GTK3 and Python3 support. I tried, and it worked on my OI deployment, so I am packaging it here.

Notes:
* As before, packaging is based on a list of files in the build/ directory - delivered by original archive with the PYC files compiled by the recipe. There is no original Makefile nor a meaningful "install" recipe, so no real sample-manifest data.
* This component is still filed into "gnome2", maybe a taxonomy change is in order. Note that despite the name, it is not related to (or constrained by) GNOME project and its versions.
* Tested to work on an OI system updated today (2020-04-14) with both the old needed dependencies for gcm-1.1.0 installed and the new ones revised for this recipe (vte => vte-291, pygobject-27 => pygobject-3-35, pygtk2-27 => dropped). I did not make a clean-lab run with the old dependency components not available.
* There is no GitHub release yet, so the recipe pulls a specific (currently happens to be the latest) commit as the source archive. This can be revised when a release gets tagged.
* On the up-side, with the new version of GCM my Midnight Commander mouse handling works again (disappeared mid-last year).

### CODE REVIEW

**Comment by pyhalov** [comment]
Can you test package in clean environment? Once done, I'll merge it.

**Comment by jimklimov** [comment]
I'll try to spin a new VM for this... so a LiveDVD, default X11/Mate install, update to today's pkg image, add this pkg from my build repo. Sounds right?

**Comment by pyhalov** [comment]
Yes.

File: components/desktop/gnome2/gnome-connection-manager/patches/gnome_connection_manager.py.TERM.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
++++ fixed/gnome_connection_manager.py	2020-04-15 12:39:05.024238615 +0000
+@@ -1145,7 +1145,10 @@
+             v.set_scrollback_lines(conf.BUFFER_LINES)
+             #if v.get_emulation() != os.getenv("TERM"):
+             #    os.environ['TERM'] = v.get_emulation()
++            ### As of Apr 2020 there are issues in OI use of xterm* in gcm
++            ### https://github.com/kuthulux/gnome-connection-manager/issues/5
+-            os.environ['TERM'] = 'xterm-256color'
++            #os.environ['TERM'] = 'xterm-256color'
++            os.environ['TERM'] = 'vt220'
```
Comment (pyhalov) [inline components/desktop/gnome2/gnome-connection-manager/patches/gnome_connection_manager.py.TERM.patch]: Does 'xterm' work?

**Comment by pyhalov** [comment]
Can you rename patches to XX-brief-description.patch? (where XX is 01, 02 and so on).

**Comment by jimklimov** [comment]
@pyhalov : tested in a clean OI VM, but had to build a special package for testing (identical but delivering `terminal/gcm` by name) to work around the protective constraints of userland-consolidation.

I had to also ensure that the build and test-deploy systems have the same `pkg image-update` level, otherwise automatically generated constraint on SUNWcs version forbids installing the built package too.

At this point, the new GCM is functional in the current OpenIndiana, so is much better than having nothing working :) There are some improvements to address however, as in any other development, so at some point this recipe would be bumped hopefully to reference an updated version and eventually a real 1.2.0 tagged release.

File: components/desktop/gnome2/gnome-connection-manager/Makefile [hunk 2]
```diff
@@ -10,34 +10,48 @@
...
-    sha256:71a128676d16153c735499cec9f0a1af1e2dc1107ed5c57cea1994eadc120c4c
+    sha256:92f0db50a31cee9bce271a3f85c269d604737edf139e0b55d52a36c0217876c1
 # Project site is gone, use a mirror for now
-#COMPONENT_ARCHIVE=	$(COMPONENT_NAME)_$(COMPONENT_VERSION)_all.tgz
+#COMPONENT_ARCHIVE=	$(COMPONENT_NAME_LONG)_$(COMPONENT_VERSION)_all.tgz
 #COMPONENT_ARCHIVE_URL=	$(COMPONENT_PROJECT_URL)/$(COMPONENT_ARCHIVE)
-COMPONENT_ARCHIVE=	$(COMPONENT_NAME)_$(COMPONENT_VERSION).orig.tar.gz
-COMPONENT_ARCHIVE_URL= http://ftp.heanet.ie/mirrors/www.getdeb.net/getdeb/ubuntu/pool/apps/g/$(COMPONENT_NAME)/$(COMPONENT_ARCHIVE)
+###COMPONENT_ARCHIVE=	$(COMPONENT_NAME_LONG)_$(COMPONENT_VERSION).orig.tar.gz
+COMPONENT_ARCHIVE=	$(COMPONENT_COMMIT_ID).zip
```
Comment (pyhalov) [inline components/desktop/gnome2/gnome-connection-manager/Makefile]: Please, name COMPONENT_ARCHIVE as $(COMPONENT_NAME)-$(COMPONENT_COMMIT_ID).zip, this determines how this file will be stored.
You can modify COMPONENT_ARCHIVE_URL to use $(COMPONENT_COMMIT_ID).zip unrelated to this.

**Comment by pyhalov** [comment]
Also, please, rename patches to XX-purpose.patch

File: components/desktop/gnome2/gnome-connection-manager/Makefile [hunk 3]
```diff
@@ -10,34 +10,55 @@
...
 
 #
-# Copyright (c) 2015 Jim Klimov
+# Copyright (c) 2015-2020 Jim Klimov
 #
 
 include ../../../../make-rules/shared-macros.mk
 
-COMPONENT_NAME=		gnome-connection-manager
+COMPONENT_NAME_LONG=	gnome-connection-manager
```
Comment (pyhalov) [inline components/desktop/gnome2/gnome-connection-manager/Makefile:18]: Why do you separate COMPONENT_NAME and COMPONENT_NAME_LONG?

Comment (jimklimov) [reply components/desktop/gnome2/gnome-connection-manager/Makefile:18]: This simplified building the alternate "gcm" named package for tests by just flipping a switch in a copy of the makefile. In the original project and its different hosting variants, the two strings are also variably used adding to the mess.

**Pull request merged**